data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ObjectReference.java

```diff
@@ -6,24 +6,26 @@
 package com.amazon.dataprepper.plugins.source;
 
 import java.util.Objects;
+import java.util.Optional;
 
 /**
  * Reference to an S3 object.
  */
 class S3ObjectReference {
     private final String bucketName;
     private final String key;
+    private final String owner;
 
-    private S3ObjectReference(final String bucketName, final String key) {
-        Objects.requireNonNull(bucketName, "bucketName must be non null");
-        Objects.requireNonNull(key, "key must be non null");
-
+    private S3ObjectReference(final String bucketName, final String key, final String owner) {
         this.bucketName = bucketName;
         this.key = key;
+        this.owner = owner;
     }
 
-    static S3ObjectReference fromBucketAndKey(final String bucketName, final String key) {
-        return new S3ObjectReference(bucketName, key);
+    static Builder bucketAndKey(final String bucketName, final String key) {
+        Objects.requireNonNull(bucketName, "bucketName must be non null");
+        Objects.requireNonNull(key, "key must be non null");
```

Comment:
Is there a reason for pulling out this validation from the constructor? The `Builder` is public at the moment and someone could unintentionally just use the builder directly instead of this function bypassing the validations.

Reply:
To fail as soon as the invalid call to build is made.

The best fix is to make the `Builder` constructor private. I will do that.